subt/main.py

```diff
@@ -476,13 +489,48 @@ def play_system_track(self):
             with EmergencyStopMonitor(self):
                 allow_virtual_flip = self.symmetric
                 self.go_straight(2.5)  # go to the tunnel entrance
-                dist = self.follow_wall(radius=self.walldist, right_wall=self.use_right_wall, timeout=self.timeout,
-                                        pitch_limit=LIMIT_PITCH, roll_limit=LIMIT_ROLL)
-                print(self.time, "Going HOME")
+                walldist = self.walldist
+                total_dist = 0.0
+                start_time = self.sim_time_sec
+                while self.sim_time_sec - start_time < self.timeout.total_seconds():
```

Comment:
If all of `self.sim_time_sec`, `start_time` and `self.timeout` are type of `timedelta`, you can drop the `.total_seconds()` part.

Reply:
`self.sim_time_sec` is integer, simulation time in seconds